src/lib.rs

```diff
@@ -249,7 +249,7 @@
 //! To construct your own simple adhoc error use the [miette!] macro:
 //! ```rust
 //! // my_app/lib/my_internal_file.rs
-//! use miette::{IntoDiagnostic, Result, WrapErr, miette};
+//! use miette::{miette, IntoDiagnostic, Result, WrapErr};
```

Comment:
It would actually be really great to expand the `miette` macro as part of this PR so that you can do stuff like:

`miette!("something went wrong", code = foo::bar::bar)`

Reply:
Yes, agree.

Reply:
This conflicts with `format!` named arguments

Reply:
My macro implementation:
```
    ($msg:literal, (code = $code:literal)? ) => {
        // Handle $:literal as a special case to make cargo-expanded code more
        // concise in the common case.
        crate::MietteDiagnostic::new($msg)
            $(.with_code($code))?
            .into()
    };
```

Reply:
Oh, nevermind, it should be `$(code = $code:literal)? `. But this really could create confusion between named argument and parameter

Reply:
Is it possible to add support for `miette!("{x} + {y}")` interpolation syntax without breaking others' code?

Reply:
It was even requested in #201

Reply:
Ahhhh that's a good point. I forgot about the format args thing!!

The way `tracing` resolves this is by treating key=value before the format strong as fields, and anything following the format string as format args. So it would look like this:

`miette!(code = foo::bar::baz, "error code from api: {code}", code = code)`

Where `code` is just a local var. 

How does that sound?

Reply:
Yep, if we want to support `format!("{x} + {y}")` thing, that moving fields before string is inevitable  anyway